packages/cli/src/oclif/commands/invoke.js

```diff
@@ -0,0 +1,685 @@
+const fs = require('node:fs');
+
+const _ = require('lodash');
+const { flags } = require('@oclif/command');
+const debug = require('debug')('zapier:invoke');
+const dotenv = require('dotenv');
+
+// Datetime related imports
+const chrono = require('chrono-node');
+const { DateTime, IANAZone } = require('luxon');
+
+const BaseCommand = require('../ZapierBaseCommand');
+const { buildFlags } = require('../buildFlags');
+const { localAppCommand } = require('../../utils/local');
+const { startSpinner, endSpinner } = require('../../utils/display');
+
+const ACTION_TYPE_PLURALS = {
+  trigger: 'triggers',
+  search: 'searches',
+  create: 'creates',
+};
+
+const ACTION_TYPES = ['auth', ...Object.keys(ACTION_TYPE_PLURALS)];
+
+const AUTH_FIELD_ENV_PREFIX = 'authData_';
+
+const NUMBER_CHARSET = '0123456789.-,';
+
+const FALSE_STRINGS = new Set([
+  'noo',
+  'no',
+  'n',
+  'false',
+  'nope',
+  'f',
+  'never',
+  'no thanks',
+  'no thank you',
+  'nul',
+  '0',
+  'none',
+  'nil',
+  'nill',
+  'null',
+]);
+
+const TRUE_STRINGS = new Set([['yes', 'yeah', 'y', 'true', 't', '1']]);
+
+const loadAuthDataFromEnv = () => {
+  return Object.entries(process.env)
+    .filter(([k, v]) => k.startsWith(AUTH_FIELD_ENV_PREFIX))
+    .reduce((authData, [k, v]) => {
+      const fieldKey = k.substr(AUTH_FIELD_ENV_PREFIX.length);
+      authData[fieldKey] = v;
+      return authData;
+    }, {});
+};
+
+const readStream = async (stream) => {
+  const chunks = [];
+  for await (const chunk of stream) {
+    chunks.push(chunk);
+  }
+  return chunks.join('');
+};
+
+const getMissingRequiredInputFields = (inputData, inputFields) => {
+  return inputFields.filter(
+    (f) => f.required && !f.default && !inputData[f.key]
+  );
+};
+
+const getMissingOptionalInputFields = (inputData, inputFields) => {
+  return inputFields.filter((f) => !f.required && !inputData[f.key]);
+};
+
+const parseDecimal = (s) => {
+  const chars = [];
+  for (const c of s) {
+    if (NUMBER_CHARSET.includes(c)) {
+      chars.push(c);
+    }
+  }
+  const cleaned = chars.join('').replace(/[.,-]$/, '');
+  return parseFloat(cleaned);
+};
+
+const parseInteger = (s) => {
+  const n = parseInt(s);
+  if (!isNaN(n)) {
+    return n;
+  }
+  return Math.floor(parseDecimal(s));
+};
```

Comment:
I tried putting a decimal in, but got a `s is not iterable` pointing to the for loop in `parseDecimal` 

Should `s` be a string in this case?

Reply:
I expect `inputData` should not contain non-string primitives to emulate the Zap editor. Added a check in 72dcad6.